pkgs/by-name/ko/kolla/package.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, python311Packages
+, fetchgit
+, bashate
+}:
+
+let
+  pythonPackages = python311Packages;
+
+in
+pythonPackages.buildPythonApplication rec {
+  pname = "kolla";
+  version = "18.1.0";
+
+  pyproject = true;
+
+  src = fetchgit {
+    url = "https://opendev.org/openstack/kolla";
+    hash = "sha256-jLD6ILihymQlWkkpGYC4OX8BKLpQurAK6Y5Xpju+QAI=";
+    rev = version;
+  };
+
+  postPatch = ''
+    substituteInPlace kolla/image/kolla_worker.py \
+      --replace "os.path.join(sys.prefix, 'share/kolla')," \
+      "os.path.join(PROJECT_ROOT, '../../../share/kolla'),"
+
+    substituteInPlace test-requirements.txt \
+      --replace "hacking>=3.0.1,<3.1.0" "hacking"
+
+    sed -e 's/git_info = .*/git_info = "${version}"/' -i kolla/version.py
+  '';
+
+  # fake version to make pbr.packaging happy
+  PBR_VERSION = version;
```

```suggestion
  # fake version to make pbr.packaging happy
  env.PBR_VERSION = version;
```

We try to leak things explicitly now.